src/technology_models/technologies/common/objective_function.jl/common_capital.jl

```diff
@@ -1 +1,63 @@
+##################################
+####### BuildCapacity Cost #######
+##################################
 
+function add_capital_cost!(
+    container::SingleOptimizationContainer,
+    ::U,
+    devices::IS.FlattenIteratorWrapper{T},
+    ::V,
+) where {T <: PSIP.SupplyTechnology, U <: BuildCapacity, V <: ContinuousInvestment}
+    for d in devices
+        capital_cost_data = PSIP.get_capital_cost(d)
+        _add_cost_to_objective!(container, U(), d, capital_cost_data, V())
+    end
+    return
+end
+
+function _add_proportional_term!(
```

Comment:
We can probably generalize this for any type `T <: InvestmentVariableType` and `U <: Technology`